spotify_dl/youtube.py

```diff
@@ -10,39 +10,54 @@
 from spotify_dl.constants import YOUTUBE_VIDEO_URL
 from spotify_dl.scaffold import log
 from json import loads
+import requests
+from lxml import html
+import re
 
 from click import secho
 
-def fetch_youtube_url(search_term, dev_key):
+def fetch_youtube_url(search_term, dev_key=None, scrape=False):
     """For each song name/artist name combo, fetch the YouTube URL
         and return the list of URLs"""
-    YOUTUBE_DEV_KEY = dev_key
-    youtube = build(YOUTUBE_API_SERVICE_NAME, YOUTUBE_API_VERSION,
-                    developerKey=YOUTUBE_DEV_KEY,
-                    cache_discovery=False)
     log.info(u"Searching for {}".format(search_term))
-    try:
-        search_response = youtube.search().list(q=search_term,
-                                                part='id, snippet').execute()
-        for v in search_response['items']:
-            if v['id']['kind'] == VIDEO:
-                log.debug("Adding Video id {}".format(v['id']['videoId']))
-                return YOUTUBE_VIDEO_URL + v['id']['videoId']
-    except HttpError as err: 
-        err_details = loads(err.content.decode('utf-8')).get('error').get('errors')
-        secho("Couldn't complete search due to following errors: ", fg='red')
-        for e in err_details:
-            error_reason = e.get('reason')
-            error_domain = e.get('domain')
-            error_message = e.get('message')
+    if scrape or not dev_key:
+        YOUTUBE_SEARCH_BASE = "https://www.youtube.com/results?search_query="
+        try:
+            response = html.fromstring(requests.get(YOUTUBE_SEARCH_BASE + search_term).content)
+            video = response.xpath("//a[contains(@class, 'yt-uix-tile-link')]/@href")
+            video_id = re.search("((\?v=)[a-zA-Z0-9_-]{4,15})", video[0]).group(0)[3:]
+            print(video_id)
+            return YOUTUBE_VIDEO_URL + video_id
+        except:
```

Comment:
Blank `except` are bad, as this will capture any and all exceptions - including the ones that are raised when Ctrl+C is pressed, so this makes it impossible to exit when the search is happening.